Code/Framework/AzCore/AzCore/Debug/PerformanceCollector.cpp

```diff
@@ -0,0 +1,260 @@
+            statisticalParams.emplace_back(UNITS, statistic->GetUnits());
+            statisticalParams.emplace_back(VARIANCE, statistic->GetVariance());
+            statisticalParams.emplace_back(STDEV, statistic->GetStdev());
+            statisticalParams.emplace_back(MOST_RECENT_SAMPLE, statistic->GetMostRecentSample());
+
+            Metrics::CompleteArgs completeArgs;
+            completeArgs.m_name = statistic->GetName();
+            completeArgs.m_cat = m_logCategory;
+            completeArgs.m_dur = AZStd::chrono::microseconds(aznumeric_cast<long long>(statistic->GetAverage()));
+            completeArgs.m_args = statisticalParams;
+            m_eventLogger.RecordCompleteEvent(completeArgs);
+        }
+    }
+
+    void PerformanceCollector::RestartPeriodicEventStamps()
+    {
+        AZStd::vector<Statistics::NamedRunningStatistic*> statistics;
+        m_statisticsManager.GetAllStatistics(statistics);
+        m_periodicEventStamps.clear();
+        for (auto statistic : statistics)
+        {
+            m_periodicEventStamps[statistic->GetName()] = {};
+        }
+    }
+
+    void PerformanceCollector::UpdateDataLogType(DataLogType newValue)
+    {
+        if (m_dataLogType == newValue)
+        {
+            return;
+        }
+
+        if (m_numberOfCaptureBatches > 0)
+        {
+            AZ_Warning(LogName, false, "%s changes to control params are rejected while data is being captured", __FUNCTION__);
+            return;
+        }
+
+        m_dataLogType = newValue;
+    }
+
+    void PerformanceCollector::UpdateFrameCountPerCaptureBatch(AZ::u32 newValue)
+    {
+        if (m_frameCountPerCaptureBatch == newValue)
+        {
+            return;
+        }
+
+        if (m_numberOfCaptureBatches > 0)
+        {
+            AZ_Warning(LogName, false, "%s changes to control params are rejected while data is being captured", __FUNCTION__);
+            return;
+        }
+
+        m_frameCountPerCaptureBatch = newValue;
+    }
+
+    void PerformanceCollector::UpdateWaitTimeBeforeEachBatch(AZStd::chrono::seconds seconds)
+    {
+        if (m_waitTimeBeforeEachBatch.count() == seconds.count())
+        {
+            return;
+        }
+
+        if (m_numberOfCaptureBatches > 0)
+        {
+            AZ_Warning(LogName, false, "%s changes to control params are rejected while data is being captured", __FUNCTION__);
+            return;
+        }
+
+        m_waitTimeBeforeEachBatch = seconds;
+    }
+
+    void PerformanceCollector::UpdateNumberOfCaptureBatches(AZ::u32 newValue)
+    {
+        if (m_numberOfCaptureBatches == newValue)
+        {
+            return;
+        }
+        if (m_numberOfCaptureBatches == 0)
+        {
+            CreateOutputJsonFile();
+        }
+        m_numberOfCaptureBatches = newValue;
+        AZ_TracePrintf(LogName, "%s updated value to %u", __FUNCTION__, m_numberOfCaptureBatches);
```

Comment:
This `AZ_TracePrintf` needs a `\n` added

Reply:
Done.